File: entites/Node.py
class Node:
    def __init__(self, x, y, deltaX, deltaY, rightBoundingBoxTop, rightBoundingBoxBottom, rightBoundingBoxLeft, rightBoundingBoxRight, leftBoundingBoxTop, leftBoundingBoxBottom, leftBoundingBoxLeft, leftBoundingBoxRight, rightChild, leftChild):
        self.x = x
        self.y = y
        self.deltaX = deltaX
        self.deltaY = deltaY
        self.rightBoundingBoxTop = rightBoundingBoxTop
        self.rightBoundingBoxBottom = rightBoundingBoxBottom
        self.rightBoundingBoxLeft = rightBoundingBoxLeft
        self.rightBoundingBoxRight = rightBoundingBoxRight
        self.leftBoundingBoxTop = leftBoundingBoxTop
        self.leftBoundingBoxBottom = leftBoundingBoxBottom
        self.leftBoundingBoxLeft = leftBoundingBoxLeft
        self.leftBoundingBoxRight = leftBoundingBoxRight
        self.rightChild = rightChild
        self.leftChild = leftChild

    def __str__(self):
        return f"{self.x=}, {self.y=}, {self.deltaX=}, {self.deltaY=}, {self.rightBoundingBoxTop=}, {self.rightBoundingBoxBottom=}, {self.rightBoundingBoxLeft=}, {self.rightBoundingBoxRight=}, {self.leftBoundingBoxTop=}, {self.leftBoundingBoxBottom=}, {self.leftBoundingBoxLeft=}, {self.leftBoundingBoxRight=}, {self.rightChild=}, {self.leftChild=}"

    __repr__ = __str__

File: cpp_code/entities/Node.cpp
#include "../headers/Node.h"
#include <cstdint>
#include <iostream>

struct Node {
    uint16_t x;
    uint16_t y;
    uint16_t deltaX;
    uint16_t deltaY;
    uint16_t rightBoundingBoxTop;
    uint16_t rightBoundingBoxBottom;
    uint16_t rightBoundingBoxLeft;
    uint16_t rightBoundingBoxRight;
    uint16_t leftBoundingBoxTop;
    uint16_t leftBoundingBoxBottom;
    uint16_t leftBoundingBoxLeft;
    uint16_t leftBoundingBoxRight;
    void* rightChild;
    void* leftChild;

    // Constructor
    Node(uint16_t x, uint16_t y, uint16_t deltaX, uint16_t deltaY, uint16_t rightBoundingBoxTop, uint16_t rightBoundingBoxBottom, uint16_t rightBoundingBoxLeft, uint16_t rightBoundingBoxRight, uint16_t leftBoundingBoxTop, uint16_t leftBoundingBoxBottom, uint16_t leftBoundingBoxLeft, uint16_t leftBoundingBoxRight)
        : x(x), y(y), deltaX(deltaX), deltaY(deltaY), rightBoundingBoxTop(rightBoundingBoxTop), rightBoundingBoxBottom(rightBoundingBoxBottom), rightBoundingBoxLeft(rightBoundingBoxLeft), rightBoundingBoxRight(rightBoundingBoxRight), leftBoundingBoxTop(leftBoundingBoxTop), leftBoundingBoxBottom(leftBoundingBoxBottom), leftBoundingBoxLeft(leftBoundingBoxLeft), leftBoundingBoxRight(leftBoundingBoxRight)
    {}

    friend std::ostream& operator<<(std::ostream& os, const Node& obj) {
        os << "Node{ ";
        os << "x: " << obj.x << " ";
        os << "y: " << obj.y << " ";
        os << "deltaX: " << obj.deltaX << " ";
        os << "deltaY: " << obj.deltaY << " ";
        os << "rightBoundingBoxTop: " << obj.rightBoundingBoxTop << " ";
        os << "rightBoundingBoxBottom: " << obj.rightBoundingBoxBottom << " ";
        os << "rightBoundingBoxLeft: " << obj.rightBoundingBoxLeft << " ";
        os << "rightBoundingBoxRight: " << obj.rightBoundingBoxRight << " ";
        os << "leftBoundingBoxTop: " << obj.leftBoundingBoxTop << " ";
        os << "leftBoundingBoxBottom: " << obj.leftBoundingBoxBottom << " ";
        os << "leftBoundingBoxLeft: " << obj.leftBoundingBoxLeft << " ";
        os << "leftBoundingBoxRight: " << obj.leftBoundingBoxRight << " ";
        os << "}";
        return os;
    }
};